Basic/APIDiorama/Assets/UseCaseSamples/ProximityChecks/Scripts/GameUI.cs

```diff
@@ -0,0 +1,74 @@
+using System.Collections;
+using TMPro;
+using UnityEngine;
+
+namespace Unity.Netcode.Samples.APIDiorama
+{
+    /// <summary>
+    /// Manages the UI of the "NetworkVariable vs RPCs" scene
+    /// </summary>
+    public class GameUI : MonoBehaviour
+    {
+        [SerializeField] TMP_Text startupLabel;
+        [SerializeField] TMP_Text controlsLabel;
+
+        void OnEnable()
```

Comment:
We should be symmetrical with OnDestroy, right now enable and destroy won't match and so you could have some callback subscription leak.
Plus, having this in Start would allow removing the coroutine, NetworkManager would be enabled already by then

Reply:
> Where's the asymmetry? We're adding and removing the same events already

You can enable disable multiple times without destroying. Which means that 5 calls to enable will register this callback 5 times. OnDestroy will unregister the callback once only.

> Is this consistent in 100% of cases?

If you check the NetworkManager's code, the singleton is set on Enable. Start always executes after first enable according to this doc https://docs.unity3d.com/Manual/ExecutionOrder.html

Reply:
I changed this in an old commit, the initialization is now in Start and there's no longer a coroutine. Thanks for the hint!